src/main/java/mods/railcraft/common/blocks/machine/gamma/TileDispenserCart.java

```diff
@@ -125,7 +127,7 @@ public void onPulse() {
                             float rx = MiscTools.getRand().nextFloat() * 0.8F + 0.1F;
                             float ry = MiscTools.getRand().nextFloat() * 0.8F + 0.1F;
                             float rz = MiscTools.getRand().nextFloat() * 0.8F + 0.1F;
-                            EntityItem item = new EntityItem(worldObj, x + rx, y + ry, z + rz, cartStack);
+                            Entity item = cartStack.getItem().createEntity(worldObj, new EntityItem(worldObj, x, y, z, cartStack), cartStack);
```

Comment:
The problem is actually [here](https://github.com/liachmodded/Railcraft/blob/4991b96cfe022c4a99642d5716f3d83b5c54ab78/src/main/java/mods/railcraft/common/blocks/machine/gamma/TileDispenserCart.java#L135). When forge replaces the vanilla `EntityItem` with mod custom item entity, it returns false, causing the slot of item stack in the cart dispenser not clearing the slot. So I jsut let Railcraft spawn the custom mod item entity at first, so that the call there will not return a confusing result.